deeplake/core/chunk/chunk_compressed_chunk.py

```diff
@@ -35,16 +36,198 @@ def __init__(self, *args, **kwargs):
         self._changed = False
         self._compression_ratio = 0.5
 
-    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True) -> float:  # type: ignore
+    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True, lengths: Optional[List[int]] = None) -> float:  # type: ignore
         self.prepare_for_write()
+        if lengths is not None:  # this is triggered only for htype == "text"
+            return self.extend_if_has_space_byte_compression_text(
+                incoming_samples, update_tensor_meta, lengths
+            )
         if self.is_byte_compression:
+            if isinstance(incoming_samples, np.ndarray):
+                return self.extend_if_has_space_byte_compression_numpy(
+                    incoming_samples, update_tensor_meta
+                )
             return self.extend_if_has_space_byte_compression(
                 incoming_samples, update_tensor_meta=update_tensor_meta
             )
         return self.extend_if_has_space_image_compression(
             incoming_samples, update_tensor_meta=update_tensor_meta
         )
 
+    def extend_if_has_space_byte_compression_text(
```

Comment:
Text is a special case, as the length of a string is not necessarily equal to its byte string length (no ascii), but on average its pretty close 1 byte/char.  

Note: we can use the same logic to ingest large lists of numpy arrays of different lengths efficiently, but is not implemented yet.